libraries/plugins/es_objects/es_objects.cpp

```diff
@@ -0,0 +1,337 @@
+         auto b = static_cast<const balance_object*>(obj);
+         PrepareBalance(b);
+      }
+   }
+}
+
+void es_objects_plugin_impl::createBulk(std::string type, std::string data, std::string id)
+{
+   bulk.push_back("{ \"index\" : { \"_index\" : \"bitshares-"+type+"\", \"_type\" : \"data\", \"_id\" : "+id+" } }");
+   bulk.push_back(data);
+}
+
+void es_objects_plugin_impl::SendBulk()
+{
+   // curl buffers to read
+   std::string readBuffer;
+   std::string readBuffer_logs;
+
+   std::string bulking = "";
+
+   bulking = boost::algorithm::join(bulk, "\n");
+   bulking = bulking + "\n";
+   bulk.clear();
+
+   struct curl_slist *headers = NULL;
+   curl_slist_append(headers, "Content-Type: application/json");
+   std::string url = _es_objects_elasticsearch_url + "_bulk";
+   curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+   curl_easy_setopt(curl, CURLOPT_POST, true);
+   curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+   curl_easy_setopt(curl, CURLOPT_POSTFIELDS, bulking.c_str());
+   curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
+   curl_easy_setopt(curl, CURLOPT_WRITEDATA, (void *)&readBuffer);
+   curl_easy_setopt(curl, CURLOPT_USERAGENT, "libcrp/0.1");
+   //curl_easy_setopt(curl, CURLOPT_VERBOSE, true);
+   curl_easy_perform(curl);
+
+   long http_code = 0;
+   curl_easy_getinfo (curl, CURLINFO_RESPONSE_CODE, &http_code);
+   if(http_code == 200) {
+      // all good, do nothing
+   }
+   else if(http_code == 429) {
+      // repeat request?
+   }
+   else {
+      // exit everything ?
+   }
+   if(_es_objects_logs) {
+      auto logs = readBuffer;
+      // do logs
+      std::string url_logs = _es_objects_elasticsearch_url + "objects_logs/data/";
+      curl_easy_setopt(curl, CURLOPT_URL, url_logs.c_str());
+      curl_easy_setopt(curl, CURLOPT_POST, true);
+      curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+      curl_easy_setopt(curl, CURLOPT_POSTFIELDS, logs.c_str());
+      curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
+      curl_easy_setopt(curl, CURLOPT_WRITEDATA, (void *) &readBuffer_logs);
+      curl_easy_setopt(curl, CURLOPT_USERAGENT, "libcrp/0.1");
+      //curl_easy_setopt(curl, CURLOPT_VERBOSE, true);
+      //ilog("log here curl: ${output}", ("output", readBuffer_logs));
+      curl_easy_perform(curl);
+
+      http_code = 0;
+      curl_easy_getinfo (curl, CURLINFO_RESPONSE_CODE, &http_code);
+      if(http_code == 200) {
+         // all good, do nothing
+      }
+      else if(http_code == 429) {
+         // repeat request?
+      }
+      else {
+         // exit everything ?
+      }
+   }
+}
+
+void es_objects_plugin_impl::PrepareProposal(const proposal_object* proposal_object)
+{
+   proposal_struct prop;
+   prop.id = proposal_object->id;
+   prop.expiration_time = proposal_object->expiration_time;
+   prop.review_period_time = proposal_object->review_period_time;
+   prop.proposed_transaction = fc::json::to_string(proposal_object->proposed_transaction);
```

Comment:
can it be done by adding the proposal objects fields `required_active_approvals`, `available_active_approvals`, `required_owner_approvals`, `available_owner_approvals` and `available_key_approvals` ?
let me know if this will do it so i can add these.

Reply:
Yes. It worth a try, although I'm not sure whether the final state can be correctly stored with current implementation, because if the proposal has no review period, the object will be removed in the same block when the proposal is approved.

Reply:
added on https://github.com/bitshares/bitshares-core/pull/500/commits/fe4e87b8a24800b3b94e589dd61cdc10600b7464